lib/.eslintrc.yaml

```diff
@@ -1,4 +1,8 @@
 rules:
+  no-restricted-syntax:
+    - error
+    - selector: "NewExpression[callee.name=/Error$/]:not([callee.name=/^(AssertionError|NghttpError)$/])"
+      message: "Use an error exported by the internal/errors module"
```

Comment:
I have a doubt now: does this override the `no-restricted-syntax` rule from our main config? If so, can we merge them somehow?

Reply:
Oh, yes, it will. When looking at this I did not see that it was only for ./lib. Activating it for the hole code base is probably not the best idea though. We could of course copy the rules that are currently in the main file but that is also not ideal...

Is there any possibility to combine this rule? That would be really great.

Reply:
@not-an-aardvark Do you have a suggestion? We would like to add a new restricted syntax to `lib` while not overriding the other ones configured in the [main config](https://github.com/nodejs/node/blob/5a4a1cba2445d71aadf4ebf4350090dfdb32d871/.eslintrc.js#L151-L177)

Reply:
Unfortunately, there's not a good way to do that at the moment. As a workaround, you could copy-paste the configuration from the parent directory, add the new entry, and put the resulting config in `lib/.eslintrc.yaml`. This would have the desired effect, but it could cause the two configs to get out of sync if someone updates `.eslintrc.yaml` later on and forgets to update the copied version in `lib/.eslintrc.yaml`.

There is some discussion on improving this in https://github.com/eslint/eslint/issues/9192.

Reply:
Thank you. I copied the config and put a comment explaining that it should be kept in sync.